Renaming `specialized` on import results in "unused import" warnings

## Compiler version

v3.3.3, v3.4.2

## Minimized code

```scala
import scala.{specialized => sp}

trait ImportUnusedSpecializedRename[@sp Int]
```

Script that I used for testing:
```sh
set -v

scala-cli clean ImportUnusedSpecializedRename.scala

scala-cli compile --server=false -j 17 -S 2.12.19 -Xlint:unused ImportUnusedSpecializedRename.scala
scala-cli compile --server=false -j 17 -S 2.13.14 -Wunused ImportUnusedSpecializedRename.scala
scala-cli compile --server=false -j 17 -S 3.3.3 -Wunused:imports ImportUnusedSpecializedRename.scala
scala-cli compile --server=false -j 17 -S 3.4.2 -Wunused:imports ImportUnusedSpecializedRename.scala
```

## Output

```
scala-cli clean ImportUnusedSpecializedRename.scala

scala-cli compile --server=false -j 17 -S 2.12.19 -Xlint:unused ImportUnusedSpecializedRename.scala
scala-cli compile --server=false -j 17 -S 2.13.14 -Wunused ImportUnusedSpecializedRename.scala
scala-cli compile --server=false -j 17 -S 3.3.3 -Wunused:imports ImportUnusedSpecializedRename.scala
-- Warning: /Users/.../ImportUnusedSpecializedRename.scala:1:26 
1 |import scala.{specialized => sp}
  |              ^^^^^^^^^^^^^^^^^
  |              unused import
1 warning found
scala-cli compile --server=false -j 17 -S 3.4.2 -Wunused:imports ImportUnusedSpecializedRename.scala
-- Warning: /Users/.../ImportUnusedSpecializedRename.scala:1:26 
1 |import scala.{specialized => sp}
  |              ^^^^^^^^^^^^^^^^^
  |              unused import
1 warning found
```

## Expectation

Scala3 should compile the above example without the "unused import" warnings.
As long as Scala2 compilers  can do.

## Additional Information

This issues causes a lot of extra warnings added up while compiling Cats for Scala3.
Historically, Cats uses `import scala.{specialized => sp}` in a bunch of files across the repository.

The Scala 3 check was intentionally skipping the type parameter, but inadvertently skipping the annotation.

The rename was not germane.

Worth adding that Scala 2:
```
$ scalac -d /tmp/sandbox -Xlint i20536.scala
i20536.scala:3: warning: type parameter Int defined in trait ImportUnusedSpecializedRename shadows class Int defined in package scala. You may want to rename your type parameter, or possibly remove it.
trait ImportUnusedSpecializedRename[@sp Int]
                                        ^
1 warning
```
Scala 3
```
$ scalac -d /tmp/sandbox -Wshadow:all i20536.scala
-- Warning: i20536.scala:3:40 ------------------------------------------------------------------------------------------
3 |trait ImportUnusedSpecializedRename[@sp Int]
  |                                    ^^^^^^^
  | Type parameter Int for trait ImportUnusedSpecializedRename shadows the type defined by class Int in package scala
1 warning found
```

--- a/compiler/src/dotty/tools/dotc/transform/CheckUnused.scala
+++ b/compiler/src/dotty/tools/dotc/transform/CheckUnused.scala
@@ -135,25 +135,22 @@ class CheckUnused private (phaseMode: CheckUnused.PhaseMode, suffix: String, _ke
     }
 
   override def prepareForDefDef(tree: tpd.DefDef)(using Context): Context =
-    unusedDataApply{ ud =>
+    unusedDataApply: ud =>
       if !tree.symbol.is(Private) then
         tree.termParamss.flatten.foreach { p =>
           ud.addIgnoredParam(p.symbol)
         }
-      import ud.registerTrivial
-      tree.registerTrivial
+      ud.registerTrivial(tree)
       traverseAnnotations(tree.symbol)
       ud.registerDef(tree)
       ud.addIgnoredUsage(tree.symbol)
-    }
 
   override def prepareForTypeDef(tree: tpd.TypeDef)(using Context): Context =
-    unusedDataApply{ ud =>
+    unusedDataApply: ud =>
+      traverseAnnotations(tree.symbol)
       if !tree.symbol.is(Param) then // Ignore type parameter (as Scala 2)
-        traverseAnnotations(tree.symbol)
         ud.registerDef(tree)
         ud.addIgnoredUsage(tree.symbol)
-    }
 
   override def prepareForBind(tree: tpd.Bind)(using Context): Context =
     traverseAnnotations(tree.symbol)

--- a/tests/warn/i20536.scala
+++ b/tests/warn/i20536.scala
@@ -1,0 +1,27 @@
+//> using options -Wunused:all
+object Anns {
+  final class S extends annotation.StaticAnnotation
+}
+
+object Main {
+  locally {
+    import Anns.*
+    class C[@S A]
+    C().toString
+  }
+  locally {
+    import Anns.S as T
+    class C[@T A]
+    C().toString
+  }
+  locally {
+    import scala.specialized as T
+    class C[@T A]
+    C().toString
+  }
+  locally {
+    import scala.specialized as T // warn
+    class C[A]
+    C().toString
+  }
+}

